backend/danswer/llm/llm_initialization.py

```diff
@@ -28,49 +30,73 @@ def load_llm_providers(db_session: Session) -> None:
     if not GEN_AI_API_KEY or DISABLE_GENERATIVE_AI:
         return
 
-    well_known_provider_name_to_provider = {
-        provider.name: provider
-        for provider in fetch_available_well_known_llms()
-        if provider.name != BEDROCK_PROVIDER_NAME
-    }
+    if GEN_AI_MODEL_PROVIDER == "custom":
+        # Validate that all required env vars are present
+        for var in (GEN_AI_LLM_PROVIDER_TYPE, GEN_AI_API_ENDPOINT, GEN_AI_MODEL_VERSION, GEN_AI_DISPLAY_NAME):
+            if not var:
+                logger.error(
+                    "Cannot auto-transition custom LLM provider due to missing env vars."
+                    "The following env vars must all be set:"
+                    "GEN_AI_LLM_PROVIDER_TYPE, GEN_AI_API_ENDPOINT, GEN_AI_MODEL_VERSION, GEN_AI_DISPLAY_NAME"
+                )
+                return None
+        llm_provider_request = LLMProviderUpsertRequest(
+            name=GEN_AI_DISPLAY_NAME,
+            provider=GEN_AI_MODEL_PROVIDER,
+            api_key=GEN_AI_API_KEY,
+            api_base=GEN_AI_API_ENDPOINT,
+            api_version=GEN_AI_API_VERSION,
+            custom_config={},
+            default_model_name=GEN_AI_MODEL_VERSION,
+            fast_default_model_name=FAST_GEN_AI_MODEL_VERSION,
+        )
 
-    if GEN_AI_MODEL_PROVIDER not in well_known_provider_name_to_provider:
-        logger.error(f"Cannot auto-transition LLM provider: {GEN_AI_MODEL_PROVIDER}")
-        return None
+    else:
```

Comment:
Note, everything inside this `else` clause is identical to the previous code (GitHub diff makes this unclear) so the behaviour is only changed when `GEN_AI_MODEL_PROVIDER` is `custom`.